exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/CompositeColumn.kt

```diff
@@ -39,8 +39,12 @@ abstract class BiCompositeColumn<C1, C2, T>(
     /** Transformation that receives the column's composite value and returns the parsed values of the underlying columns. */
     val transformFromValue: (T) -> Pair<C1?, C2?>,
     /** Transformation that receives the retrieved values of [column1] and [column2] and returns a composite value. */
-    val transformToValue: (Any?, Any?) -> T
+    val transformToValue: (Any?, Any?) -> T,
```

Comment:
Is it intentional to define `transformToValue` as `(Any?, Any?) -> T`. Does it mean that the user should manually call `columnType.valueFromDB()` inside that function?

I tried to replace it with `val transformToValue: (C1?, C2?) -> T?` and update `restoreValueFromParts()` to 
```
override fun restoreValueFromParts(parts: Map<Column<*>, Any?>): T? {
  val v1 = parts[column1]?.let { column1.columnType.valueFromDB(it) }
  val v2 = parts[column2]?.let { column2.columnType.valueFromDB(it) }
  val result = transformToValue(v1, v2)
  check(result != null || nullable) {
      "Null value received from DB for non-nullable ${this::class.simpleName} column"
  }
  return result
}
```

and it seems to be working, but I don't know which use cases I loast with such update. But after fixing `CompositeMoneyColumn` I also got green tests (without modifying tests itself).

Anyway the current variant is also ok to me, but `(Any, Any)` surrounded by generics looks alien)